.github/workflows/cd.yml

```diff
@@ -0,0 +1,68 @@
+name: Continuous Deployment
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group:  ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  cd:
+    name: CD
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        lfs: true
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.8'
+
+    - name: Cache dependency
+      id: cache-dependencies
+      uses: actions/cache@v4
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .[dev]
+
+    - name: Install and run pre-commit hooks
+      run: |
+        pre-commit install
+        pre-commit run --all-files --show-diff-on-failure --color=always
+
+    - name: Check types with pyright
+      run: |
+        pyright sbi
+
+    - name:  Run the fast and the slow tests with coverage
+      run: |
+        pip install pytest-cov
+        pytest -m "not gpu" tests/ --cov=sbi --cov-report=xml --exitfirst
```

Comment:
Small note: `-n auto` from #1050 will probably not end up in both files

Reply:
Oh that's a nice new change. I added it "back" in